### PR TITLE
Update hubspot plugin

### PR DESCRIPTION
**Pull request by bozhi**
- Update coercing function use correct lists for the various object types
  - Do not support custom property names for now because they will just fail right now
  - Add hubspot_create_association function

### CODE REVIEW

File: hubspot/plugin.py [hunk 1]
```diff
@@ -617,8 +630,6 @@ def hubspot_update_contacts(
     Contact IDs that have been updated.
 
 The property names MUST be one of the following:
-
-Default Properties of Type String:
```
Comment (jngiam) [inline hubspot/plugin.py:621]: let's keep these, i suspect code gen will find them helpful.

e.g., when to call .as_datetime etc

File: hubspot/plugin.py [hunk 2]
```diff
@@ -1002,9 +1013,12 @@ def hubspot_search_contacts(
...
     filters = []
     for property_name, value in search_criteria.items():
-        filters.append(
-            {"propertyName": property_name, "operator": "EQ", "value": value}
-        )
+        if value:
+            filters.append(
+                {"propertyName": property_name, "operator": "EQ", "value": value}
+            )
+    if not filters:
```
Comment (jngiam) [inline hubspot/plugin.py:1020]: add a comment on why we are doing this check

File: hubspot/plugin.py [hunk 3]
```diff
@@ -1361,8 +1381,6 @@ def hubspot_update_companies(
 Company IDs that have been updated.
 
 The property names to update MUST be one of the following:
-
-Default Properties of Type String:
```
Comment (jngiam) [inline hubspot/plugin.py:1365]: ditto

File: hubspot/plugin.py [hunk 4]
```diff
@@ -2317,3 +2343,83 @@ def hubspot_fetch_associated_object_ids(
...
+    target_object_type: Union[HubSpotObjectType, HubSpotCustomObjectType],
+    target_object_id: str,
+) -> List[str]:
+    """
+    Returns the IDs of target objects associated with the source object
+    using the HubSpot association API. You must use this to find HubSpot
+    objects that are associated to each other.
+    """
+    source_type_name = _HUBSPOT_OBJECT_TYPE_IDS.get(
+        source_object_type.name, source_object_type.name
```
Comment (jngiam) [inline hubspot/plugin.py:2385]: can you add a comment on why returning itself if not found is the right thing to do?

File: hubspot/plugin.py [hunk 5]
```diff
@@ -2317,3 +2343,83 @@ def hubspot_fetch_associated_object_ids(
...
+def hubspot_create_association_between_object_ids(
+    association_type: HubSpotAssociationType,
+    source_object_type: Union[HubSpotObjectType, HubSpotCustomObjectType],
+    source_object_id: str,
+    target_object_type: Union[HubSpotObjectType, HubSpotCustomObjectType],
+    target_object_id: str,
+) -> List[str]:
+    """
+    Returns the IDs of target objects associated with the source object
+    using the HubSpot association API. You must use this to find HubSpot
```
Comment (jngiam) [inline hubspot/plugin.py:2381]: find -> create

also what does this return?

should it just returned number of associations created?

**Review by jngiam — APPROVED** [review]

File: hubspot/plugin.py (hunk 2 above)
Comment (bozhi) [reply hubspot/plugin.py:1028]: Added comment below this check

File: hubspot/plugin.py (hunk 4 above)
Comment (bozhi) [reply hubspot/plugin.py]: Updated to not support CustomProperty

File: hubspot/plugin.py (hunk 5 above)
Comment (bozhi) [reply hubspot/plugin.py]: Updated doc.

**Pull request merged**